test/test_puma_server.rb

```diff
@@ -1087,4 +1087,80 @@ def test_accept_econnaborted
     stub_accept_nonblock(error)
     assert_empty @events.stderr.string
   end
+
+  def test_integer_response_header_name
+    server_run app: ->(env) { [200, { 1 => 'Boo'}, []] }
+    data = send_http_and_read "GET / HTTP/1.0\r\n\r\n"
+    assert_match(/HTTP\/1.1 500 Internal Server Error/, data)
+  end
+
+  def test_nil_response_header
+    server_run app: ->(env) { [200, nil, []] }
+    data = send_http_and_read "GET / HTTP/1.0\r\n\r\n"
+    assert_match(/HTTP\/1.1 500 Internal Server Error/, data)
+  end
+
+  # The header must not contain a Status key.
+  def test_response_header_name_is_status
+    server_run app: ->(env) { [200, {'Status' => '500'}, []] }
+    data = send_http_and_read "GET / HTTP/1.0\r\n\r\n"
+
+    assert_match(/HTTP\/1.1 500 Internal Server Error/, data)
+  end
+
+  def test_response_header_value_is_integer
+    server_run app: ->(env) { [200, {'Content-Length' => 500}, []] }
+    data = send_http_and_read "GET / HTTP/1.0\r\n\r\n"
+
+    assert_match(/HTTP\/1.0 200 OK\r\nContent-Length: 500\r\n\r\n/, data)
+  end
+
+  #Special headers starting “rack.” are for communicating with the server, and must not be sent back to the client.
```

Comment:
Not 100% necessary, but it would be cool if you added a comment like this to every test